specification/src/chapter4.adoc

```diff
@@ -2,69 +2,897 @@
+
+| CAT_NNN
+| Local sealing for a TA, or a TEE guest, MUST be unique to TEE domain and to a physical instance of a system.
+
+| CAT_NNN
+| Local sealing for a TA, or a TEE guest, SHOULD also be unique to the TEE guest or the TA.
+
+| CAT_NNN
+| Local sealing MAY be layered.
+
+|===
+
+For example:
+
+* TEE domain unique sealing keys derived by a RoT from a hardware unique key
+* TA, or guest TEE, unique sealing keys derived by TEE OS or SPM from a TEE domain unique sealing key
+
+==== Device access control
+
+For the purpose of this specification, a device can be a logical device. A physical device can present one or more logical devices, each with its own (logical) control interface. 
+
+The security guarantees also apply to device initiated accesses, for example DMA and interrupts. 
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN
+| A static partition TEE MUST use IOPMP to enforce access rules for devices.
+
+| CAT_NNN
+| A virtualized TEE MUST use IOMTT and IOMMU to enforce access rules for devices assigned to Normal or TEE domains, and SHOULD use IOPMP to enforce access rules for Root devices.
+
+|===
+
+For a static partition TEE, domain level granularity can be sufficient as device access within TEE and Normal domains is governed by TEE OS and the rich OS respectively. It can be implemented using IOPMP. Policy can be controlled by boot configuration, by a HW or FW RoT.
+
+For a virtualized TEE, IOMTT enforces supervisor domain level access rules (physical isolation). IOMMU enforces guest and TA level access rules (virtualization), supporting device assignment to a guest TEE or a TA.  
+
+NOTE: IOMTT can also be sufficient for protecting Root devices in the sense that M-mode can enforce that its own resources are never assigned to another domain. Use of IOPMP or similar still adds further protections. For example, a system may require that Root devices cannot be used to access memory assigned to Confidential domain.
+
+==== System integration
+
+In the case of a Global Platform TEE system a rich OS in Normal domain is free to schedule services, including TEE services, on any Hart available to it. The number and make-up of supervisor domains can be known, and a simple convention can be used for common identification (SDID value, see xref:chapter3.adoc#_supervisor_domains[supervisor domains]) of Normal, TEE, and Root domains across multiple Harts in a system. 
+
+System integration in this context involves providing _security attributes_ on a system interconnect, tagging all transactions (CPU or system agent initiated) to either Root, Normal, or TEE domains. 
+
+Possible use cases include:
+
+* Tweaking cryptographic memory protection (uniqueness)
+* Tagging interrupts, debug accesses, or coherent memory accesses
+* Device assignment (IOPMP/IOMTT integration), static or dynamic
+
+For interoperability, an implementation could adopt the following convention commonly used in Global Platform TEE ecosystems:
```

Comment:
is this a bit too arm ? also limits to single TEE domain. ( comment below talks about extending granularity)

Reply:
It probably is too Arm. I will remove it. I was debating it with myself even when writing it :-)

Like you say, that particular encoding only applies to systems that try to interoperate with the Arm version of GP TEE. And they simply follow Arm specifications for this.

There is nothing in either GP TEE or RISC-V supervisor domains that prevent people from going beyond Arm.